extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/security/PersistentLoginManager.java

```diff
@@ -0,0 +1,118 @@
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.security.identity.SecurityIdentity;
+import io.vertx.core.http.Cookie;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * A class that manages persistent logins.
+ *
+ * This is done by encoding an expiry time, and the current username into an encrypted cookie
+ *
+ * TODO: make this pluggable
+ */
+public class PersistentLoginManager {
+
+    private static final Logger log = Logger.getLogger(PersistentLoginManager.class);
+
+    private final SecretKey secretKey;
+    private final String cookieName;
+    private final long timeoutMillis;
+
+    public PersistentLoginManager(String encryptionKey, String cookieName, long timeoutMillis) {
+        try {
+            this.cookieName = cookieName;
+            this.timeoutMillis = timeoutMillis;
+            if (encryptionKey == null) {
+                secretKey = KeyGenerator.getInstance("AES").generateKey();
+            } else if (encryptionKey.length() < 16) {
+                throw new RuntimeException("Shared keys for persistent logins must be more than 16 characters long");
+            } else {
+                MessageDigest sha256 = MessageDigest.getInstance("SHA-256");
+                sha256.update(encryptionKey.getBytes(StandardCharsets.UTF_8));
+                secretKey = new SecretKeySpec(sha256.digest(), "AES");
+            }
+        } catch (Exception t) {
+            throw new RuntimeException(t);
+        }
+
+    }
+
+    public RestoreResult restore(RoutingContext context) {
+        Cookie existing = context.getCookie(cookieName);
+        if (existing == null) {
+            return null;
+        }
+        String val = existing.getValue();
+        try {
+            Cipher cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
+            cipher.init(Cipher.DECRYPT_MODE, secretKey);
+            String result = new String(cipher.doFinal(Base64.getDecoder().decode(val)), StandardCharsets.UTF_8);
+            int sep = result.indexOf(":");
+            if (sep == -1) {
+                return null;
+            }
+            long expire = Long.parseLong(result.substring(0, sep));
+            if (System.currentTimeMillis() > expire) {
+                return null;
+            }
+            return new RestoreResult(result.substring(sep + 1), (System.currentTimeMillis() - expire) > 1000 * 60); //new cookie every minute
+        } catch (Exception e) {
+            log.debug("Failed to restore persistent user session", e);
+            return null;
+        }
+    }
+
+    public void save(SecurityIdentity identity, RoutingContext context, RestoreResult restoreResult) {
+        if (restoreResult != null) {
+            if (!restoreResult.newCookieNeeded) {
+                return;
+            }
+        }
+        try {
+            Cipher cipher = Cipher.getInstance("AES/ECB/PKCS5Padding");
+
+            StringBuilder contents = new StringBuilder();
+            //TODO: do we need random padding?
```

Comment:
Can you take a look at this change set https://github.com/pedroigor/quarkus/commit/fb0b13d36d1adae9da4398c4fda8c69a5baed20c ?

It is about using AES GCM so that we have another random value to encrypt blocks. I have also changed to 128 which I think is a enough and help to balance with performance. We can make this configurable too.

I'm not using association data as that would require us to keep some state. But would be nice in order to make sure the user presenting the cookie is in fact the one to which the cookie was set. But it does not compromise the security.